src/item_group.cpp

```diff
@@ -74,8 +74,12 @@ item Single_item_creator::create_single( const time_point &birthday, RecursionLi
     if( modifier ) {
         modifier->modify( tmp );
     } else {
+        int qty = item::INFINITE_CHARGES;
+        if( modifier ) {
+            qty = rng( modifier->charges.first, modifier->charges.second );
+        }
```

Comment:
This code is unreachable as you have already verified `modifier` is `false` in outer `if` (line 74).